src/main/java/io/anserini/kg/freebase/IndexTopic.java

```diff
@@ -0,0 +1,184 @@
+    public AtomicLong indexedDocuments = new AtomicLong();
+  }
+
+  /**
+   * Program arguments to hold parameters and properties.
+   */
+  private Args args;
+
+  private final Path indexPath;
+  private final Path collectionPath;
+  private final Counters counters;
+
+  /**
+   * Constructor
+   *
+   * @param args program arguments
+   * @throws Exception
+   */
+  public IndexTopic(Args args) throws Exception {
+
+    // Copy arguments
+    this.args = args;
+
+    // Log parameters
+    LOG.info("Collection path: " + args.input);
+    LOG.info("Index path: " + args.index);
+
+    // Initialize variables
+    this.indexPath = Paths.get(args.index);
+    if (!Files.exists(this.indexPath)) {
+      Files.createDirectories(this.indexPath);
+    }
+
+    collectionPath = Paths.get(args.input);
+    if (!Files.exists(collectionPath) || !Files.isReadable(collectionPath)) {
+      throw new IllegalArgumentException("Document file/directory " + collectionPath.toString() +
+              " does not exist or is not readable, please check the path");
+    }
+
+    this.counters = new Counters();
+  }
+
+  private void run() throws IOException, InterruptedException {
+    final long start = System.nanoTime();
+    LOG.info("Starting indexer...");
+
+    final Directory dir = FSDirectory.open(indexPath);
+    final SimpleAnalyzer analyzer = new SimpleAnalyzer();
+    final IndexWriterConfig config = new IndexWriterConfig(analyzer);
+    config.setOpenMode(IndexWriterConfig.OpenMode.CREATE);
+    config.setCodec(new Lucene62Codec(Lucene50StoredFieldsFormat.Mode.BEST_SPEED));
+    config.setUseCompoundFile(false);
+
+    final IndexWriter writer = new IndexWriter(dir, config);
+    index(writer, collectionPath);
+
+    int numIndexed = writer.maxDoc();
+    try {
+      writer.commit();
+    } finally {
+      try {
+        writer.close();
+      } catch (IOException e) {
+        LOG.error(e);
+      }
+    }
+
+    LOG.info("Indexed documents: " + counters.indexedDocuments.get());
+    final long durationMillis = TimeUnit.MILLISECONDS.convert(System.nanoTime() - start, TimeUnit.NANOSECONDS);
+    LOG.info("Total " + numIndexed + " documents indexed in " +
+            DurationFormatUtils.formatDuration(durationMillis, "HH:mm:ss"));
+  }
+
+  /**
+   * Run the indexing process.
+   *
+   * @param writer index writer
+   * @param inputFile which file to index from the collection
+   * @throws IOException on error
+   */
+  private void index(IndexWriter writer, Path inputFile) throws IOException {
+    TopicLuceneDocumentGenerator transformer = new TopicLuceneDocumentGenerator();
+    transformer.config(args);
+    transformer.setCounters(counters);
+
+    int cnt = 0;
+    ObjectTriplesIterator iter = new ObjectTriplesIterator(inputFile);
+    while (iter.hasNext()) {
+      ObjectTriples d = iter.next();
+      // make a Topic from the ObjectTriples
+      String subject = d.getSubject();
+      Topic topic = new Topic(subject);
```

Comment:
Can you explain why we need a separate "Topic" object here?
Can't we just take an ObjectTriples and generate the proper Lucene document?
In other words, does it make sense that the "Object" needs to have an independent existence?